File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> counter;

        for(int num : nums){
            counter[num]++;
        }

        for(auto entry : counter){
            if(entry.second != 1){
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = Counter(nums) #count all freq of each num
        freq = [[] for i in range(len(nums) + 1)] #number of elements that appear index times

        for num, count in count.items():
            freq[count].append(num)
        
        ans = []
        for i in range(len(freq) - 1, -1, -1):
            for j in range(len(freq[i])):
                ans.append(freq[i][j])
                if len(ans) == k:
                    return ans


        